### ISSUE TITLE
`g_memdup` is deprecated and `g_memdup2` should be used instead

### ISSUE THREAD

**Issue by ggardet** [nnstreamer/nnstreamer#3218]
# Issue Description
`g_memdup` is deprecated and `g_memdup2` should be used instead.
```
[   69s] /home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/gst/nnstreamer/tensor_converter/tensor_converter.c:749:9: error: 'g_memdup' is deprecated: Use 'g_memdup2' instead [-Werror=deprecated-declarations]
[   69s]   749 |         data = g_memdup (info.data + idx, size);
[   69s]       |         ^~~~
[   69s] In file included from /usr/include/glib-2.0/glib.h:82,
[   69s]                  from /usr/include/gstreamer-1.0/gst/gst.h:27,
[   69s]                  from /home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/gst/nnstreamer/tensor_converter/tensor_converter.h:37,
[   69s]                  from /home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/gst/nnstreamer/tensor_converter/tensor_converter.c:48:
[   69s] /usr/include/glib-2.0/glib/gstrfuncs.h:257:23: note: declared here
[   69s]   257 | gpointer              g_memdup         (gconstpointer mem,
[   69s]       |                       ^~~~~~~~
[   69s] cc1: all warnings being treated as errors
```

Expected Result
============
Compilation should pass.

How to Reproduce
===============
1. Try to build against latest glib (2.68.1)
2. See the warning/error

Further Information
===============
* Exact OS version : openSUSE Tumbleweed

**Comment by myungjoo** [comment]
@again4you PTAL or Reassign. Tizen uses glib 2.62, but it is expected to be upgraded soon (after 6.5 M1 release)

**Comment by myungjoo** [comment]
Anyway, g_memdup2 is "Since: 2.68". Thus, the fix should be backward-compatible: let's define a macro somewhere.

**Comment by myungjoo** [comment]
> @again4you PTAL or Reassign. Tizen uses glib 2.62, but it is expected to be upgraded soon (after 6.5 M1 release)

Nah.. forget it. I'll apply fix. :)

**Comment by myungjoo** [comment]
> # Issue Description
> `g_memdup` is deprecated and `g_memdup2` should be used instead.
> 
> ```
> [   69s] /home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/gst/nnstreamer/tensor_converter/tensor_converter.c:749:9: error: 'g_memdup' is deprecated: Use 'g_memdup2' instead [-Werror=deprecated-declarations]
> [   69s]   749 |         data = g_memdup (info.data + idx, size);
> [   69s]       |         ^~~~
> [   69s] In file included from /usr/include/glib-2.0/glib.h:82,
> [   69s]                  from /usr/include/gstreamer-1.0/gst/gst.h:27,
> [   69s]                  from /home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/gst/nnstreamer/tensor_converter/tensor_converter.h:37,
> [   69s]                  from /home/abuild/rpmbuild/BUILD/nnstreamer-1.7.1/gst/nnstreamer/tensor_converter/tensor_converter.c:48:
> [   69s] /usr/include/glib-2.0/glib/gstrfuncs.h:257:23: note: declared here
> [   69s]   257 | gpointer              g_memdup         (gconstpointer mem,
> [   69s]       |                       ^~~~~~~~
> [   69s] cc1: all warnings being treated as errors
> ```
> 
> # Expected Result
> Compilation should pass.
> 
> # How to Reproduce
> 1. Try to build against latest glib (2.68.1)
> 2. See the warning/error
> 
> # Further Information
> * Exact OS version : openSUSE Tumbleweed

Looks like we no longer use g_memdup.

https://github.com/nnstreamer/nnstreamer/commit/0cb379f723f1551d38ee297519f19938194b404d#diff-4c7f564ff9b3dcc0d5bf728c30be7ecf305dfc30731cd543c93b7703742c3de1

**Comment by myungjoo** [comment]
Because this issue is already fixed with 0cb379f, I'm closing the issue.

**Comment by ggardet** [comment]
It seems we also need https://github.com/nnstreamer/nnstreamer/commit/85c7fe1ad940a60f91a40c8a502042fb12461124 which is not in v2.0.0 but only in `main`.
